llama-index-integrations/vector_stores/llama-index-vector-stores-vertexaivectorsearch/llama_index/vector_stores/vertexaivectorsearch/base.py

```diff
@@ -0,0 +1,280 @@
+        region: Optional[str] = None,
+        index_id: Optional[str] = None,
+        endpoint_id: Optional[str] = None,
+        gcs_bucket_name: Optional[str] = None,
+        credentials_path: Optional[str] = None,
+        text_key: str = DEFAULT_TEXT_KEY,
+        **kwargs: Any,
+    ) -> "VertexAIVectorStore":
+        """Create VertexAIVectorStore from config."""
+        return cls(
+            project_id=project_id,
+            region=region,
+            index_name=index_id,
+            endpoint_id=endpoint_id,
+            gcs_bucket_name=gcs_bucket_name,
+            credentials_path=credentials_path,
+            text_key=text_key,
+            **kwargs,
+        )
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "VertexAIVectorStore"
+
+    @property
+    def client(self) -> Any:
+        """Get client."""
+        return self._index
+
+    @property
+    def index(self) -> Any:
+        """Get client."""
+        return self._index
+
+    @property
+    def endpoint(self) -> Any:
+        """Get client."""
+        return self._endpoint
+
+    @property
+    def staging_bucket(self) -> Any:
+        """Get client."""
+        return self._staging_bucket
+
+    def add(
+        self,
+        nodes: List[BaseNode],
+        is_complete_overwrite: bool = False,
+        **add_kwargs: Any,
+    ) -> List[str]:
+        """Add nodes to index.
+
+        Args:
+            nodes: List[BaseNode]: list of nodes with embeddings
+
+        """
+        ids = []
+        embeddings = []
+        metadatas = []
+        for node in nodes:
+            node_id = node.node_id
+            metadata = node_to_metadata_dict(
+                node, remove_text=False, flat_metadata=False
+            )
+            embedding = node.get_embedding()
+
+            ids.append(node_id)
+            embeddings.append(embedding)
+            metadatas.append(metadata)
+
+        data_points = utils.to_data_points(ids, embeddings, metadatas)
+        # self._document_storage.add_documents(list(zip(ids, nodes)))
+
+        if self._stream_update:
+            utils.stream_update_index(index=self._index, data_points=data_points)
+        else:
+            if self._staging_bucket is None:
+                raise ValueError(
+                    "To update a Vector Search index a staging bucket must"
+                    " be defined."
+                )
+            utils.batch_update_index(
+                index=self._index,
+                data_points=data_points,
+                staging_bucket=self._staging_bucket,
+                is_complete_overwrite=is_complete_overwrite,
+            )
+        return ids
+
+    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
+        """
+        Delete nodes using with ref_doc_id.
+
+        Args:
+            ref_doc_id (str): The doc_id of the document to delete.
+
+        """
+        self._index.remove_datapoints(datapoint_ids=[ref_doc_id])
```

Comment:
This is actually meant to be a filtered delete -- deleting any point with `ref_doc_id==ref_doc_id` in the metadata

Is this possible to do here?

Reply:
Yes. Needs two operations. Updated with filtered delete based on `ref_doc_id`